### PR TITLE
Add keyboard shortcuts

### LINKED ISSUE #145: Add keyboard navigation
I would like to add some keyboard navigation options to dino. This is the plan (^A means &lt;Primary&gt;A or Ctrl+A):

- [ ] Menu entries
  - Application menu
    - ^A Accounts
    - ^P Settings
    - ^Q Quit
  - Add menu
    - ^N Start Chat
    - ^M Join Conference
  - ^F Find
  - ^O Open / attach
  - ^L List occupants
  - Alt+E Encryption menu
  - "Hamburger menu"
    - Alt+Enter Contact Details
- [ ] Others
  - ^PageUp / ^PageDown and ^Tab / ^Shift+Tab to switch conversations (#142)
  - Alt+1, Alt+2, ... to switch to first, second, ... conversation (counted from the bottom / the oldest to make switching more consistent if new conversations are added)
  - PageUp / PageDown to scroll (#143)
  - ^W to close the current conversation
- [x] Just for reference:
  - F10 to select menu buttons, Tab / Shift+Tab to cycle through the buttons
  - unknown: select application menu

I will implement it in the next days. Feel free to comment.

(As a side note: It should be possible to see the keys in the menu popups if this feature request gets implemented: https://bugzilla.gnome.org/show_bug.cgi?id=787342)

### PR DESCRIPTION
**Pull request by sandwm**
* Add a Gtk.ShortcutsWindow where the user can see the shortcuts.
 * Add Dino.Ui.Util.Shortcuts which retrieves the shortcuts from the
   ShortcutsWindow and provides a utility function to create actions.
 * Add a bunch of shortcuts, using the above.

Fixes #145, #142, #143

### CODE REVIEW

File: main/src/ui/conversation_selector/list.vala [hunk 1]
```diff
@@ -147,6 +189,21 @@ public class List : ListBox {
...
+            if (filter((ListBoxRow) row))
+                visible_rows.append((ListBoxRow) row);
+        }
+        return visible_rows;
+    }
+
+    private void select_visible_row(uint nr) {
+        GLib.List<ListBoxRow> rows = get_visible_rows();
+        if (rows.length() >= nr)
+            rows.nth(rows.length() - nr).data.activate();
```
Comment (mar-v-in) [inline main/src/ui/conversation_selector/list.vala]: This makes no sense IMHO: the last rows are those whose last message is most in the past. These are usually those you don't contact rather often, so why put a shortcut on them?

**Comment by sandwm** [comment]
Thanks for the comments. I will probably get back to this next week. (Might have some further notes.)

Should I append the changes to this pull request or create a new one with less changing back-and-forth?

**Comment by fiaxh** [comment]
You can add the changes to this pull request if you want. They can be squashed later.

**Comment by fiaxh** [comment]
@sandwm May I assume that you won't have the time to incorporate mar-v-in's feedback? If you should have any questions, you can also contact us via the Dino MUC at `chat@dino.im`.

**Pull request closed without merging**